include/glow/Base/Tensor.h

```diff
@@ -157,6 +161,10 @@ class Tensor final {
   /// \returns the number of bytes required to store the tensor.
   uint64_t getSizeInBytes() const { return type_.getSizeInBytes(); }
 
+  /// \returns the TensorPool managing this object, or nullptr if it is
```

Comment:
Why is this a part of Tensor?  I assumed that the tensor pool would manage some tensors. Just like the Placeholder binding is managing some other resources. The fact that the tensor is managed by some pool should not matter to the tensor, just like a car that is managed by valet should not have special features to support the valet - just a key.

Reply:
The PlaceholderBindings owns its Tensors - and will delete them on `clear()`, `erase()` and the destructor. This seemed the least disruptive option but there are some alternatives we could consider:
* Change Placeholder so it takes a `Tensor *` rather than `Tensor &&` doesn't own its placeholders and calling code must go through pairs() and delete or `reclaim()` as appropriate. Adds complexity to all usage of PlaceholderBindings and we're likely to forget.
* Change the TensorPool to return a new Tensor, which is deleted but reclaims the buffer in its destructor. We'd need to subclass Tensor and have the TensorPool return a new ManagedTensor* backed by a shared buffer, and have to change PlaceholderBindings to take a Tensor* rather than a Tensor&&. This will be slightly less efficient than what we have now, primarily mainly because we're no longer reusing the actual Tensor objects (just the backing memory) and we'll be hitting malloc regularly anyway. We'd also be putting all access to Tensor through a vtable.
* Change Tensor replacing isUnowned with a refcount over the backing buffer and have the TensorPool always hold a refount. Bit of juggling here to make sure that the TensorPool only returns buffers with a single refcount since it wont get a clear signal when the buffer isn't used anymore. A bit more complicated dealing with cases where the Tensor really doesn't own its buffer (such as the input buffers in Onnxifi).
* Change PlaceholderBindings so that it has another map of managed tensors (and which pool manages them). Basically the same as this except the additional code is in PlaceholderBindings instead of Tensor. The PlaceholderBindings would need to look up multiple maps to do anything though, potentially.

I prefer the small change to Tensor over any of these, but willing to be convinced.

Reply:
Okay.